pkg/systemd/services/timer-dialog.jsx

```diff
@@ -353,15 +506,26 @@
                                         </>}
                                         {repeat == "yearly" && <>
                                             <DatePicker aria-label={_("Pick date")}
-                                                        buttonAriaLabel={_("Toggle date picker")}
-                                                        locale={timeformat.dateFormatLang()}
-                                                        weekStart={timeformat.firstDayOfWeek()}
-                                                        onChange={(str, data) => {
-                                                            const arr = [...repeatPatterns];
-                                                            arr[idx].date = str;
-                                                            setRepeatPatterns(arr);
-                                                        }}
-                                                        appendTo={() => document.body} />
+                                                inputProps={{
+                                                    id: repeat,
+                                                    pattern: '^[0-9]{4}[-](0[1-9]|1[0-2])[-](0[1-9]|[12][0-9]|3[0-1])',
+                                                    isRequired: true,
+                                                    validated: repeatPatterns[idx].validated,
+                                                    onBlur: event => {
+                                                        repeatValidate(event);
+                                                    },
+                                                    onChange: str => {
+                                                        const arr = [...repeatPatterns];
+                                                        arr[idx].date = str;
+                                                        setRepeatPatterns(arr);
+                                                    }
+                                                }}
+                                                appendTo={() => document.body}
+                                                buttonAriaLabel={_("Toggle date picker")}
+                                                locale={timeformat.dateFormatLang()}
+                                                weekStart={timeformat.firstDayOfWeek()}
+                                                // isRequired // No Required flag for DatePicker
+                                            />
```

Comment:
These 20 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)